lxd/images.go

```diff
@@ -1995,7 +1996,74 @@ func imageRefresh(d *Daemon, r *http.Request) Response {
 	return OperationResponse(op)
 }
 
-func imageSyncBetweenNodes(d *Daemon, req api.ImagesPost, fingerprint string) error {
+func autoSyncImagesTask(d *Daemon) (task.Func, task.Schedule) {
+	f := func(ctx context.Context) {
+		opRun := func(op *operation) error {
+			return autoSyncImages(ctx, d)
+		}
+
+		op, err := operationCreate(d.cluster, "", operationClassTask, db.OperationImagesSynchronize, nil, nil, opRun, nil, nil)
+		if err != nil {
+			logger.Error("Failed to start image synchronization operation", log.Ctx{"err": err})
+			return
+		}
+
+		logger.Infof("Synchronizing images across the cluster")
+		_, err = op.Run()
+		if err != nil {
+			logger.Error("Failed to synchronize images", log.Ctx{"err": err})
+			return
+		}
+		logger.Infof("Done synchronizing images across the cluster")
+	}
+
+	return f, task.Daily()
+}
+
+func autoSyncImages(ctx context.Context, d *Daemon) error {
+	// In order to only have one task operation executed per image when syncing the images
+	// across the cluster, only leader node can launch the task, no others.
+	localAddress, err := node.ClusterAddress(d.db)
+	if err != nil {
+		return err
+	}
+	leader, err := d.gateway.LeaderAddress()
+	if err != nil {
+		return err
+	}
+	if localAddress != leader {
```

Comment:
This check should go into `autoSyncImagesTask` otherwise every LXD system will start logging messages like:
 - Synchronizing images across the cluster
 - Done synchronizing images across the cluster

Which will confuse our users, both those that don't have image sync configured and those that aren't even using clustering. You can see how we're doing this kind of check for other tasks already like the automated snapshots, backup expiry and snapshot expiry cases.